src/test/java/utility/Utility.java

```diff
@@ -15,49 +15,41 @@ public Utility(){
 
     }
 
-    //Nice try, think harder...
-   /* public void goToPage(Object object){
-
-        if(object instanceof LoginPage){
-
-            System.out.println("I am login page");
-            //driver.navigate().to(FileReaderManager.getInstance().getConfigFileReader().getApplicationURL().concat("login"));
-
-        } *//* else if (object instanceof SecurePage) {
-
-            System.out.println("I am secure page");
-
-        } else {
-            System.out.println("I am not a page");
-        } *//*
-
-    }*/
-
 
     /* Page Actions */
 
+    /* Clicks a clickable web element */
     public void clickAction(WebElement webElement){
         webElement.click();
     }
 
-    public void enterData(WebElement webElement, String data){
-        webElement.sendKeys(data);
+    /* Enters data into the field */
+    public void enterData(WebElement inputField, String data){
+        inputField.sendKeys(data);
     }
 
-    public void selectDropdownValue(WebElement element, String value){
-        Select select = new Select(element);
-        select.selectByValue(value);
+    /* Selects the dropdown menu item based on the text displayed in the dropdown*/
+    public void selectDropdownByText(WebElement dropdown, String text){
+        Select select = new Select(dropdown);
+
+        for (int i = 0; i < select.getOptions().size() ; i++) {
+            if(select.getOptions().get(i).getText().equals(text)){
+                select.getOptions().get(i).click();
+            }
+        }
     }
 
-    public void clickCheckbox(List<WebElement> list, String num) {
+    /* Selects a checkbox by accepting the number of the checkbox as they are shown on the page
+    * eg: 1 for the first one, 2 for the one below it, etc.  */
+    public void clickCheckboxByOrder(List<WebElement> checkboxList, String num) {
 
         int checkboxNumber = Integer.parseInt(num) - 1;
 
-        list.get(checkboxNumber).click();
+        checkboxList.get(checkboxNumber).click();
 
     }
 
-    //Login function
```

Comment:
Login function needs to be in Login Page class - dont pass Web elements to login function there, just values but use utility methods to enter input fields and click on login button